clarify dev readmes

- [ ] in all 3 readmes, let's be more clear about how to get started. such as which node version, and in which order to deal with the 3 subpackages

e.g. say one should
```
npm install -g n
n 16.15.1
```

- [ ] in https://github.com/epfml/disco/blob/develop/browser/README.md , the following is confusing/contradicting

`Once you have the server running, and built discojs, you may start the browser.

⚠ WARNING: discojs
Since discojs is used as a package in the browser (and this in turn is written in TypeScript) it is required that you not build discojs, npm install is not enough.`


- [ ] getting another error when running the helper server locally with `npm run dev`

```
> server@1.0.0 dev
> nodemon src/run_server.ts

[nodemon] 2.0.16
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: ts,json
[nodemon] starting `node --inspect=0.0.0.0:9229 --nolazy -r ts-node/register src/run_server.ts`
Debugger listening on ws://0.0.0.0:9229/7f0c10ae-6489-455b-b8a4-de5b4ba3f04b
For help, see: https://nodejs.org/en/docs/inspector
/Users/mljaggi/git/disco/server/node_modules/ts-node/src/index.ts:820
    return new TSError(diagnosticText, diagnosticCodes);
           ^
TSError: ⨯ Unable to compile TypeScript:
src/router/federated.ts:341:51 - error TS2339: Property 'weights' does not exist on type 'typeof import("/Users/mljaggi/git/disco/discojs/dist/serialization")'.

341     const serializedWeights = await serialization.weights.encode(weights)
                                                      ~~~~~~~
src/router/federated.ts:358:35 - error TS2339: Property 'weights' does not exist on type 'typeof import("/Users/mljaggi/git/disco/discojs/dist/serialization")'.

358     const weights = serialization.weights.decode(rawWeights)
                                      ~~~~~~~

    at createTSError (/Users/mljaggi/git/disco/server/node_modules/ts-node/src/index.ts:820:12)
    at reportTSError (/Users/mljaggi/git/disco/server/node_modules/ts-node/src/index.ts:824:19)
    at getOutput (/Users/mljaggi/git/disco/server/node_modules/ts-node/src/index.ts:1014:36)
    at Object.compile (/Users/mljaggi/git/disco/server/node_modules/ts-node/src/index.ts:1322:43)
    at Module._compile (/Users/mljaggi/git/disco/server/node_modules/ts-node/src/index.ts:1454:30)
    at node:internal/modules/cjs/loader:1159:10
    at Object..ts (/Users/mljaggi/git/disco/server/node_modules/ts-node/src/index.ts:1458:12)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19) {
  diagnosticCodes: [ 2339, 2339 ]
}
[nodemon] app crashed - waiting for file changes before starting...
```

--- a/browser/README.md
+++ b/browser/README.md
@@ -1,36 +1,33 @@
 # Browser
 
-This folder contains the (Vue3) code for the disco browser
+This folder contains the (Vue 3) code for the Disco browser client
 
-## How to run
+## Preliminaries
 
-To run the browser locally we need two things:
+To run the Disco client locally we need two things:
 
 1. [Server](../server/README.md)
 2. [discojs](../discojs/README.md)
 
 Please refer to their respective readmes in order to understand what is required before going to the next step.
 
-Once you have the server running, and *built* discojs, you may start the browser.
+⚠ Once you have the server running, and *built* discojs, you may start the browser.
 
 > **⚠ WARNING: discojs**  
-> Since `discojs` is used as a package in the browser (and this in turn is written in TypeScript) it is required that you not build `discojs`, `npm install` is not enough.
+> Since `discojs` is used as a package in the browser (and this in turn is written in TypeScript) it is required that you build `discojs`, `npm install` is not enough.
 
-## Running the browser
+## Hosting the Disco client
 
-### Node Installation and NPM installation
+### Node and NPM Installation
 
-The app is running under Node 15.12.0. It can be downloaded from [here](https://nodejs.org/en/download/releases/).
+The app is running under Node.js v16. NPM is a package manager for the JavaScript runtime environment Node.js.
+We recommend using [nvm](https://github.com/nvm-sh/nvm) for installing both Node.js and NPM.
 
-NPM is a package manager for the JavaScript runtime environment Node.js.  
-To start the application (running locally) run the following command.  
-Note: the application is currently developed using [NPM 7.6.3](https://www.npmjs.com/package/npm/v/7.6.3).
+To install the application's dependencies, run the following commands:
 
 ```
-npm install
+npm ci
 ```
-
-This command will install the necessary libraries required to run the application (defined in the `package.json` and `package-lock.json`). The latter command is only required when one is using the app for the first time.
 
 > **⚠ WARNING: Apple Silicon.**  
 > `TensorFlow.js` in version `3.13.0` currently supports for M1 mac laptops. However, make sure you have an `arm` node executable installed (not `x86`). It can be checked using:
@@ -39,17 +36,21 @@ This command will install the necessary libraries required to run the applicatio
 node -p "process.arch"
 ```
 
-### Compiling and hot-reload for development
+### Running for development
 
-To launch the application run the following command:
+The Disco client is a Vue app. For development purposes, we recommend running the client in Vue's development mode:
 
 ```
 npm run serve
 ```
 
-This will start the application locally with two visualization options:
+which supports hot-reload. This will start the application locally with two visualization options:
 
 1. One can access the running app locally, with a `localhost link`
 2. One can access the running app on any device that has access to the network of his machine. To do so, use the `network link`.
 
->
+Note: As mentioned above, the browser requires the [server](../browser/README.md) to run. Make sure both services run on different ports.
+
+### Running for production
+
+TODO

--- a/browser/README.md
+++ b/browser/README.md
@@ -2,23 +2,23 @@
 
 This folder contains the (Vue 3) code for the Disco browser client
 
-## Preliminaries
+## Prerequisites
 
-To run the Disco client locally we need two things:
+Before running the Disco client locally we need to install two things:
 
-1. [Server](../server/README.md)
-2. [discojs](../discojs/README.md)
+1. [discojs](../discojs/README.md)
+2. [server](../server/README.md)
 
-Please refer to their respective readmes in order to understand what is required before going to the next step.
+Please refer to their respective linked readmes in order to understand what is required before going to the next step.
 
-⚠ Once you have the server running, and *built* discojs, you may start the browser.
+⚠ Once you have the server running, and *built* discojs, you may start the browser, as we describe below.
 
 > **⚠ WARNING: discojs**  
 > Since `discojs` is used as a package in the browser (and this in turn is written in TypeScript) it is required that you build `discojs`, `npm install` is not enough.
 
-## Hosting the Disco client
+## Hosting the Disco browser client
 
-### Node and NPM Installation
+### Installation
 
 The app is running under Node.js v16. NPM is a package manager for the JavaScript runtime environment Node.js.
 We recommend using [nvm](https://github.com/nvm-sh/nvm) for installing both Node.js and NPM.
@@ -30,11 +30,13 @@ npm ci
 ```
 
 > **⚠ WARNING: Apple Silicon.**  
-> `TensorFlow.js` in version `3.13.0` currently supports for M1 mac laptops. However, make sure you have an `arm` node executable installed (not `x86`). It can be checked using:
+> `TensorFlow.js` since version `3.13.0` and newer do support M1 processors for macs. To do so, make sure you have an `arm` node executable installed (not `x86`). It can be checked using:
 
 ```
 node -p "process.arch"
 ```
+
+which should return `arm64`.
 
 ### Running for development
 
@@ -49,7 +51,7 @@ which supports hot-reload. This will start the application locally with two visu
 1. One can access the running app locally, with a `localhost link`
 2. One can access the running app on any device that has access to the network of his machine. To do so, use the `network link`.
 
-Note: As mentioned above, the browser requires the [server](../browser/README.md) to run. Make sure both services run on different ports.
+Note: As mentioned above, the browser requires a running helper [server](../browser/README.md) to function. Make sure both services are running on different ports.
 
 ### Running for production
 

--- a/discojs/README.md
+++ b/discojs/README.md
@@ -2,7 +2,7 @@
 
 discojs contains the core code of disco.
 
-## Node and NPM installation
+## Installation
 
 The app is running under Node.js v16. NPM is a package manager for the JavaScript runtime environment Node.js.
 We recommend using [nvm](https://github.com/nvm-sh/nvm) for installing both Node.js and NPM.
@@ -14,7 +14,7 @@ npm ci
 ```
 
 > **⚠ WARNING: Apple Silicon.**
-> `TensorFlow.js` in version `3.13.0` currently supports for M1 mac laptops. However, make sure you have an `arm` node executable installed (not `x86`). It can be checked using:
+> `TensorFlow.js` since version `3.13.0` and newer do support M1 processors for macs. To do so, make sure you have an `arm` node executable installed (not `x86`). It can be checked using:
 
 ```
 node -p "process.arch"

--- a/discojs/README.md
+++ b/discojs/README.md
@@ -2,31 +2,32 @@
 
 discojs contains the core code of disco.
 
-## Node Installation and NPM installation
+## Node and NPM installation
 
-The app is running under Node 15.12.0. It can be downloaded from [here](https://nodejs.org/en/download/releases/).
+The app is running under Node.js v16. NPM is a package manager for the JavaScript runtime environment Node.js.
+We recommend using [nvm](https://github.com/nvm-sh/nvm) for installing both Node.js and NPM.
 
-NPM is a package manager for the JavaScript runtime environment Node.js.  
-To start the application (running locally) run the following command.  
-Note: the application is currently developed using [NPM 7.6.3](https://www.npmjs.com/package/npm/v/7.6.3).
+To install the application's dependencies, run the following command:
 
 ```
-npm install
+npm ci
 ```
 
-This command will install the necessary libraries required to run the application (defined in the `package.json` and `package-lock.json`). The latter command is only required when one is using the app for the first time.
-
-> **⚠ WARNING: Apple Silicon.**  
+> **⚠ WARNING: Apple Silicon.**
 > `TensorFlow.js` in version `3.13.0` currently supports for M1 mac laptops. However, make sure you have an `arm` node executable installed (not `x86`). It can be checked using:
 
 ```
 node -p "process.arch"
 ```
 
+which should return `arm64`.
+
 ## Build
 
-In order to enable the Browser to use the `discojs` package, we must build discojs:
+In order to enable the browser and server modules to use the `discojs` package, we must build discojs:
 
 ```
 npm run build
 ```
+
+This invokes the TypeScript compiler (`tsc`). To recompile from stratch, simply `rm -rf dist/` before running `npm run build` again.

--- a/server/README.md
+++ b/server/README.md
@@ -7,7 +7,7 @@ For federated learning, the helper server receives all weight updates (but never
 
 ### Requirements
 
-The server is running as a single ExpressJS app. It mainly requires [Node](https://nodejs.org/en/), [Express](https://expressjs.com/), [PeerServer](https://github.com/peers/peerjs-server) and [Tensorflow](https://www.tensorflow.org/js).
+The helper server is running as a single ExpressJS app. It mainly requires [Node](https://nodejs.org/en/), [Express](https://expressjs.com/), [PeerServer](https://github.com/peers/peerjs-server) and [Tensorflow](https://www.tensorflow.org/js).
 
 To install the dependencies, run
 
@@ -15,7 +15,7 @@ To install the dependencies, run
 npm ci
 ```
 
-This server also requires the [discojs](../discojs/README.md) package. Make sure `discojs` is *built* before proceeding to the next steps.
+This server also requires the [discojs](../discojs/README.md) package. Make sure `discojs` is *built* before proceeding to the next steps, by following the [discojs readme](../discojs/README.md).
 
 ### Running the server locally
 
@@ -70,10 +70,10 @@ Deployment files:
 
 To change the GAE app configuration, you can modify the file `app.yaml`.
 
-To deploy the app on GAE, you can run the following command, where deai-313515 is the current PROJECT-ID:
+To deploy the app on GAE, you can run the following command, where disco-313515 is the current PROJECT-ID:
 
 ```
-gcloud app deploy --project=deai-313515 app.yaml --version prod
+gcloud app deploy --project=disco-313515 app.yaml --version prod
 ```
 
 :exclamation: Important!
@@ -105,7 +105,7 @@ In the `tsconfig` we add the base esm module that we use as well as including th
 
 tl;dr: `tsconfig.json` specifies the general setup for ts (including testing), `tsconfig.prod.json` adds production specific commands to `tsconfig.json`.
 
-## DeAI Helper Server, for decentralized training
+## Disco Helper Server, for decentralized training
 
 ### Components
 
@@ -118,9 +118,9 @@ The PeerServer stores the entire list of connected peers, which the peers need t
 
 #### ML Tasks
 
-The list of ML training tasks available to DeAI clients are provided by this helper server. Their descriptions as well as their deep learning model architectures must be made available to peers, which is achieved via the following routing paths:
+The list of ML training tasks available to disco clients are provided by this helper server. Their descriptions as well as their deep learning model architectures must be made available to peers, which is achieved via the following routing paths:
 
-- `/tasks`: JSON file containing meta-data (including task id) on all available DeAI training tasks
+- `/tasks`: JSON file containing meta-data (including task id) on all available training tasks
 - `/tasks/task_id/{model.json, weights.bin}`: Tensorflow neural network model files for the given task id (model architecture & initialization weights)
 
 Tasks are stored in `tasks.json`. The models are declared in `models.js`.
@@ -132,7 +132,7 @@ Adding a new task server-side can easily be done by following the next steps:
 - add an entry to `tasks.json` with the task's parameters
 - add a `createModel` function to `models.js` with the task's model architecture and export it
 
-## DeAI Helper Server, for federated training
+## Disco Helper Server, for federated training
 
 ### Components
 
@@ -179,9 +179,9 @@ The federated learning server keeps track of connected clients and weights from 
 
 #### ML Tasks
 
-The list of ML training tasks available to DeAI clients are provided by this helper server. Their descriptions as well as their deep learning model architectures must be made available to all clients, which is achieved via the following `GET` request routes.
+The list of ML training tasks available to clients is provided by this helper server. Their descriptions as well as their deep learning model architectures must be made available to all clients, which is achieved via the following `GET` request routes.
 
-- JSON file containing meta-data (including task id) on all available FeAI training tasks
+- JSON file containing meta-data (including task id) on all available training tasks
   ```
   URL: /tasks
   ```

--- a/server/README.md
+++ b/server/README.md
@@ -7,7 +7,13 @@ For federated learning, the helper server receives all weight updates (but never
 
 ### Requirements
 
-The server is running as a single ExpressJS app. It mainly requires [Node](https://nodejs.org/en/), [Express](https://expressjs.com/), [PeerServer](https://github.com/peers/peerjs-server) and [Tensorflow](https://www.tensorflow.org/js). All library requirements are included in the `package.json` file.
+The server is running as a single ExpressJS app. It mainly requires [Node](https://nodejs.org/en/), [Express](https://expressjs.com/), [PeerServer](https://github.com/peers/peerjs-server) and [Tensorflow](https://www.tensorflow.org/js).
+
+To install the dependencies, run
+
+```
+npm ci
+```
 
 ### Running the server locally
 

--- a/server/README.md
+++ b/server/README.md
@@ -15,6 +15,8 @@ To install the dependencies, run
 npm ci
 ```
 
+This server also requires the [discojs](../discojs/README.md) package. Make sure `discojs` is *built* before proceeding to the next steps.
+
 ### Running the server locally
 
 From this folder, you can run the server on localhost:8080 with `npm run dev`. This runs via the `nodemon` package, so it automatically restarts the process after changes.

